File: algorithms/670.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int maximumSwap(int num) {
        string snum = to_string(num);
        int n = snum.size();
        for (int i = 0; i < n; i++) {
            int t = n-1;
            for (int j = n-1; j > i; j--)
                if (snum[j] > snum[t])
                    t = j;
            if (snum[i] < snum[t]) {
                swap(snum[i], snum[t]);
                break;
            }
        }
        return stoi(snum);
    }
};

int main() {
    return 0;
}

File: algorithms/python3/670.py
class Solution:
    def maximumSwap(self, num):
        """
        :type num: int
        :rtype: int
        """
        s = list(str(num))
        for i in range(len(s)):
            for j in range(i):
                t = s[:]
                t[i], t[j] = t[j], t[i]
                num = max(num, int(''.join(t)))
        return num
